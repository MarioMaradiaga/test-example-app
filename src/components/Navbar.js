import { Link } from "react-router-dom";

function Navbar({ isAdmin }) {
  return (
    <nav className="navbar">
      <Link to="/home">Home</Link>
      {isAdmin && <Link to="/settings">Admin Settings</Link>}
    </nav>
  );
}

export default Navbar;
