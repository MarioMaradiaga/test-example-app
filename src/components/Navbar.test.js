import Navbar from "./Navbar";
import renderer from "react-test-renderer";

jest.mock("react-router-dom", () => {
  return { Link: () => "Link" };
});
test("should render Navbar", () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render admin link for admins", () => {
  const tree = renderer.create(<Navbar isAdmin />).toJSON();
  expect(tree).toMatchSnapshot();
});
