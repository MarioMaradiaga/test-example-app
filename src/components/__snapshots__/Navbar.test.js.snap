// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Navbar 1`] = `
<nav
  className="navbar"
>
  Link
</nav>
`;

exports[`should render admin link for admins 1`] = `
<nav
  className="navbar"
>
  Link
  Link
</nav>
`;
